docs/specification.md

```diff
@@ -1,153 +1,207 @@
-
-### Types
-
-#### SignedMEVPayloadHeader
+- result: `null`
+- error: code and message set in case an exception happens while getting the payload.
 
-See [here](https://github.com/ethereum/consensus-specs/blob/v1.1.10/specs/phase0/beacon-chain.md#custom-types) for the definition of fields like `BLSSignature`
+#### Specification
+1. Builder software **MUST** verify `signature` is valid under `publicKey`.
+2. Builder software **MUST** respond to requests where `timestamp` is before the latest announcement from the validator or more than +/- 10 seconds from the current time with `-32006: Invalid timestamp`.
+3. Builder software **MUST** store `feeRecipient` in a map keyed by `publicKey`.
 
-- message: [MEVPayloadHeader](#mevpayloadheader)
-- signature: BLSSignature
+### `builder_getHeaderV1`
 
-#### MEVPayloadHeader
+#### Request
 
-- payloadHeader: [`ExecutionPayloadHeaderV1`](https://github.com/ethereum/consensus-specs/blob/v1.1.10/specs/bellatrix/beacon-chain.md#executionpayloadheader)
-- feeRecipientDiff: Quantity, 256 Bits - the change in balance of the feeRecipient address
+- method: `builder_getHeaderV1`
+- params:
+  1. `hash`: `DATA`, 32 Bytes - Hash of block which the validator intends to use as the parent for its proposal.
 
-#### SignedBlindedBeaconBlock
+#### Response
 
-See [here](https://github.com/ethereum/consensus-specs/blob/v1.1.10/specs/phase0/beacon-chain.md#custom-types) for the definition of fields like `BLSSignature`
+- result: `object`
+    - `payload`: [`ExecutionPayloadHeaderV1`](#executionpayloadheaderv1).
+    - `value`: `DATA`, 32 Bytes - the change in wei balance of the `feeRecipient` account.
+    - `publicKey`: `DATA`, 48 Bytes - the public key associated with the builder.
+    - `signature`: `DATA`, 96 Bytes - BLS signature of the builder over `payload` and `value`.
+- error: code and message set in case an exception happens while getting the payload.
 
-- message: [BlindedBeaconBlock](#blindedbeaconblock)
-- signature: BLSSignature
+#### Specification
+1. Builder software **SHOULD** respond immediately with the `payload` that increases the `feeRecipient`'s balance by the most.
+2. Builder software **MUST** return `-32001: Unknown hash` if the block identified by `hash` does not exist.
+3. Builder software **MUST** return `-32002: Unknown validator` if the recovered validator public key has not been mapped to a `feeRecipient`.
 
-#### BlindedBeaconBlock
+### `builder_getPayloadV1`
 
-This is forked from [here](https://github.com/ethereum/consensus-specs/blob/v1.1.10/specs/phase0/beacon-chain.md#beaconblock) with `body` replaced with `BlindedBeaconBlockBody`
+#### Request
 
-- slot: Slot
-- proposer_index: ValidatorIndex
-- parent_root: Root
-- state_root: Root
-- body: BlindedBeaconBlockBody
+- method: `builder_getPayloadV1`
+- params:
+  1. `block`: `DATA`, arbitray length - SSZ encoded [`BlindBeaconBlock`](#blindbeaconblock).
+  2. `signature`: `DATA`, 96 Bytes - BLS signature of the validator over `block`.
```

Comment:
Another option is to pass in `SignedBlindBeaconBlock` which is basically 1. + 2.

Have we considered adding `publicKey` as an input? We don't need it if the builder has the beacon state and the ability to look up the `block. proposer_index`'s public key

Reply:
As mentioned in this [comment](https://github.com/flashbots/mev-boost/pull/82/files#r845003459), I am making the assumption that builder's have the ability to look up the `block.proposer_index`'s public key. I think this is a reasonable and important assumption, because without it, the builder may receive may `builder_getHeader` requests with different `feeRecipient`s, which will cause them to calculate blocks that have no possibility of being included.

I'm not necessarily opposed to passing `SignedBlindBeaconBlock` - the rationale for not doing so currently is it breaks from the current pattern of `signature` always being an RPC parameter / response.

Reply:
No worries. I think `block` and `signature` are great. Let's see if others have feedback on this